scripts/veloxbench/veloxbench/_benchmark.py

```diff
@@ -0,0 +1,228 @@
+import datetime
+import functools
+import json
+import logging
+import os
+import subprocess
+from typing import Any, Dict, List, Optional, Tuple
+
+import conbench.runner
+
+logging.basicConfig(format="%(levelname)s: %(message)s")
+
+
+def _now_formatted() -> str:
+    now = datetime.datetime.now(datetime.timezone.utc)
+    return now.isoformat()
+
+
+def github_info(velox_info: Dict[str, Any]) -> Dict[str, Any]:
+    return {
+        "repository": "https://github.com/facebookincubator/velox",
+        "commit": velox_info["velox_git_revision"],
+    }
+
+
+def velox_info() -> Dict[str, Any]:
+    # TODO: undo the hard coding with ways to get the values from velox
+    # c.f. https://github.com/ursacomputing/benchmarks/blob/033eee0951adbf41931a2de95caccbac887da6ff/benchmarks/_benchmark.py#L30-L48
+    return {
+        "velox_version": "fake-version-0.0.1",
+        "velox_compiler_id": None,
+        "velox_compiler_version": None,
+        "velox_compiler_flags": None,
+        # This should work, though it would be much better to get the sha from velox directly so we know we're using the right one.
+        "velox_git_revision": subprocess.check_output(["git", "rev-parse", "HEAD"])
+        .decode("ascii")
+        .strip(),
+    }
+
+
+class Benchmark(conbench.runner.Benchmark):
+    arguments = []
+    options = {"cpu_count": {"type": int}}
+
+    def __init__(self):
+        super().__init__()
+
+    @functools.cached_property
+    def velox_info(self) -> Dict[str, Any]:
+        return velox_info()
+
+    @functools.cached_property
+    def velox_info_r(self) -> Dict[str, Any]:
+        return self._get_velox_info_r()
+
+    @functools.cached_property
+    def github_info(self) -> Dict[str, Any]:
+        return github_info(self.velox_info)
+
+    def benchmark(
+        self,
+        f,
+        extra_tags: Dict[str, Any],
+        options: Dict[str, Any],
+        case: Optional[tuple] = None,
+    ):
+        cpu_count = options.get("cpu_count", None)
+        if cpu_count is not None:
+            # TODO: how to set multithreading?
+            raise NotImplementedError()
```

Comment:
I presume the default is multithreaded + that's what we really care about so we can leave this alone, but if we wanted to be able to tune that parameter, we would add something in here

Reply:
All current microbenchmarks are single-threaded and very short-lived. Why do we care about multi-threading here?

Reply:
Yeah, you're absolutely right for here we don't really care for the micro benchmarks (and thanks for the correction on these specifics). But we will (probably?) want to run the macrobenchmarks multithreaded when we get to those. As it stands, this code path shouldn't be triggered, so I'm inclined to leave it in (with a comment about using it for macrobenchmarks)